enhancements/console/quick-starts.md

```diff
@@ -0,0 +1,154 @@
+- [ ] Graduation criteria for dev preview, tech preview, GA
+
+## Open Questions [optional]
+
+## Summary
+
+OpenShift's Serverless team has proposed an idea to create a "Quick Starts"
+mechanism which introduces users to various ways of interacting with serverless
+in the Console. Quick Starts should be a mechanism we can use to vastly improve
+our customer's initial user experience on a empty cluster or with all various 
+workflows:
+
+The goal of this proposal is to define a lightweight mechanism for OpenShift's
+Console component, to guide users thought various workflows, and help them
+understand the steps neccesary to get the desired outcome:
+
+* Install operator
+* Deployment of showcase application
+* Cluster settings
+* Networking
+* ...
+
+For Quick Starts we need a mechanism for their creation and publishment.
+
+## Motivation
+
+Help users with their understanding the principles of their workflows by guiding them though the necessary steps.
+
+### Goals
+
+1. Provide a mechanism to display user guides for various workflows.
+2. Make users understand what are the steps needed to achieve their goals.
+3. Provide new CRD format for writing the Quick Starts.
+4. Have a repository for the out-of-the-box Quick Starts, those that describe how to install an operator or go though standard workflow.
+
+## Proposal
+
+* Introduce CRD format that will be used for writing Quick Starts.
+* Introduce default repository for Quick Starts.
+  
+### User Stories
+
+#### Story 1
+
+As an administrator of an OpenShift cluster, I need a guide to walk me through how to install OpenShift Serverless modules (Serving and Eventing) in a cluster.
+
+#### Story 2
+
+As an administrator of an OpenShift cluster, I need a guide to walk me through how to update an OpenShift cluster.
+
+#### Story 3
+
+As a developer, I need a guide to walk me through how to deploy an existing application as a serverless workload.
+
+#### Story 4
+
+As a operator creator I want to provide operator consumers with a guide on how to install and user the my operator.
+
+### Implementation Details
+
+1. In order to provide mechanism to discribe a Quick Start, new CRD named `QuickStarts` will be created.
+2. A new `openshift/quick-tours` repository will be created which will contain all supported Quick Starts. The `openshift/quick-tours` repository will have `release-*` branches.
+3. `console-operator` will import all the existing Quick Starts CRs from appropriate branch of the `openshift/quick-starts` repository into the `/manifest` directory, so that the CVO can:
+   * create them if the CR doesn't exists.
+   * update the CRs upon the cluster update (since CVO is doing `apply`).
+4. `console-operator` will be resposible for removing older `QuickStarts` versions, based on their CRs `spec.version` field that will be in semver format. If the semver version range won't match the cluster version the `console-operator` will remove the CR.
```

Comment:
Is this necessarily? It seems like we can handle versioning by having release-* branches in the quick-starts repository and just vendoring the right version. I'm not sure we need a version range in the quick start resource itself.

Reply:
So field was mainly added, so the operator can get rid of old QuickStarts, since after the cluster update they wont be deleted from the cluster.
Also I think it cant hurt, for the QuickStart CR to know whats it's version.

Reply:
Hey @jhadvig. I'm still unclear. The new quick start for the current version should just replace the old one since they have the same name. In that case, the operator doesn't need to do anything.

If we need to remove a quick start entirely, I think we might have to hard-code that special case in the operator for one release. I wouldn't expect this to happen often.

Reply:
ok! makes sense, will remove.